File: leetcode/top-interview-150/26/Java/Solution.java
public class Solution {

    public static int removeDuplicates(int[] nums) {
        int k = 0;
        int n = nums.length;
        for (int i = 0; i < n; i++) {
            if (nums[i] != nums[k]) {
                k++;
                nums[k] = nums[i];
            }
        }
        return k + 1;
    }

    public static void main(String[] args) {
        int[] nums = { 0, 0, 1, 1, 1, 2, 2, 3, 3, 4 };
        System.out.println(removeDuplicates(nums));
    }
}

File: leetcode/top-interview-150/26/Solution.go
package main

import "fmt"

func main() {
	nums := []int{1, 1, 2}
	removeDuplicates(nums)
}

func removeDuplicates(nums []int) int {
	k := 0
	for _, num := range nums {
		if num != nums[k] {
			k++
			nums[k] = num
		}
	}

	fmt.Print(nums)

	fmt.Print(k + 1)

	return k + 1
}
